File: implementations/python/sdk/src/multiplexed_ai_sdk/contracts/control/execution_control_state.py
from dataclasses import dataclass

from ...wire_model import AiSdkWireModel
from .execution_control_action import AiSdkExecutionControlAction
from .execution_control_status import AiSdkExecutionControlStatus


@dataclass(frozen=True)
class AiSdkExecutionControlState(AiSdkWireModel):
    status: AiSdkExecutionControlStatus
    pending_action: AiSdkExecutionControlAction
    updated_at_utc: str
    reason: str | None = None
    waiting_key: str | None = None
    waiting_step_name: str | None = None
    pause_requested_at_utc: str | None = None
    paused_at_utc: str | None = None
    resume_requested_at_utc: str | None = None
    input_received_at_utc: str | None = None

File: implementations/node/sdk/src/contracts/control/execution-control-state.ts
import type { AiSdkExecutionControlAction } from "./execution-control-action.js";
import type { AiSdkExecutionControlStatus } from "./execution-control-status.js";

export interface AiSdkExecutionControlState {
  readonly status: AiSdkExecutionControlStatus;
  readonly pendingAction: AiSdkExecutionControlAction;
  readonly reason?: string;
  readonly waitingKey?: string;
  readonly waitingStepName?: string;
  readonly updatedAtUtc: string;
  readonly pauseRequestedAtUtc?: string;
  readonly pausedAtUtc?: string;
  readonly resumeRequestedAtUtc?: string;
  readonly inputReceivedAtUtc?: string;
}
